### ISSUE TITLE
We should have Cloud SDK version check in the plugin

### ISSUE THREAD

**Issue by YnR** [GoogleCloudPlatform/google-cloud-eclipse#668]
When trying to deploy the Hello World to my project, this is what I get:

<img width="377" alt="screen shot 2016-09-06 at 10 46 22 am" src="https://cloud.githubusercontent.com/assets/1758462/18284674/c1496ba4-741f-11e6-982b-65ba9dcbf715.png">
1. The View more detail being the same as what is above is ... a little frustrating.
2. Not clear what went wrong or how to solve it

Console output:

```
********************************************************
There is a new version of the SDK available.
-----------
Latest SDK:
Release: 1.9.42
Timestamp: Fri Jul 15 11:45:47 PDT 2016
API versions: [1.0]

-----------
Your SDK:
Release: 1.9.34
Timestamp: Thu Feb 11 11:36:15 PST 2016
API versions: [1.0]

-----------
Please visit https://developers.google.com/appengine/downloads for the latest SDK.
********************************************************
Reading application configuration data...


Beginning interaction for module default...
0% Scanning for jsp files.
Success.
Temporary staging for module default directory left in /Users/ynr/Documents/workspace/.metadata/.plugins/com.google.cloud.tools.eclipse.appengine.deploy/tmp/1473183949120/staging
Usage: gcloud [optional flags] <group | command>
  group may be           auth | beta | components | compute | config | container
                         | dataproc | deployment-manager | dns | preview |
                         projects | source | sql | topic
  command may be         docker | feedback | help | info | init | version

The *gcloud* CLI manages authentication, local configuration, developer
workflow, and interactions with the Google Cloud Platform APIs.

commonly used flags:
  --account ACCOUNT      Google Cloud Platform user account to use for
                         invocation.
  --configuration CONFIGURATION
                         The configuration to use for this command invocation.
  --format FORMAT        The format for printing command output resources.
  --help                 Display detailed help.
  --project PROJECT_ID   Google Cloud Platform project ID to use for this
                         invocation.
  --quiet, -q            Disable all interactive prompts.
  --verbosity VERBOSITY  Override the default verbosity for this command.  This
                         must be a standard logging verbosity level: [debug,
                         info, warning, error, critical, none] (Default:
                         [warning]).
  -h                     Print a summary help and exit.
  -v, --version          Print version information and exit. This flag is only
                         available at the global level.

other flags:
  Run `gcloud --help`
  for the full list of available flags for this command.

command groups:
  auth                   Manage oauth2 credentials for the Google Cloud SDK.
  beta                   *(BETA)* Beta versions of gcloud commands.
  components             List, install, update, or remove Google Cloud SDK
                         components.
  compute                Read and manipulate Google Compute Engine resources.
  config                 View and edit Cloud SDK properties.
  container              Deploy and manage clusters of machines for running
                         containers.
  dataproc               Create and manage Google Cloud Dataproc clusters and
                         jobs.
  deployment-manager     Manage deployments of cloud resources.
  dns                    Manage your Cloud DNS managed-zones and record-sets.
  preview                Manage Preview CLI command groups.
  projects               Manage your Projects.
  source                 Cloud git repository commands.
  sql                    Manage Cloud SQL databases.
  topic                  gcloud supplementary help.

commands:
  docker                 Provides the docker CLI access to the Google Container
                         Registry.
  feedback               Provide feedback to the Google Cloud SDK team.
  help                   Prints detailed help messages for the specified
                         commands.
  info                   Display information about the current gcloud
                         environment.
  init                   Initialize or reinitialize gcloud.
  version                Print version information for Cloud SDK components.


For more detailed information on this command and its flags, run:
  gcloud --help

ERROR: (gcloud) Invalid choice: 'app'.

Valid choices are [auth, beta, components, compute, config, container, dataproc, deployment-manager, dns, docker, feedback, help, info, init, meta, preview, projects, source, sql, topic, version].

```

**Comment by akerekes** [comment]
@YnR the version of cloudsdk is too old, please run `gcloud components update` and try again.

We should have a version check in the plugin...

**Comment by elharo** [comment]
Or in appengine-plugins-core

**Comment by elharo** [comment]
FYI, step 1 of fixing this is

https://github.com/GoogleCloudPlatform/appengine-plugins-core/issues/165

Once that's done, we can address the problem in Eclipse.

**Comment by jeffdahl** [comment]
Just installed the latest cloud SDK . . . eclipse states "Cloud SDK too old".
```
jeff@page:~$ gcloud -v
Google Cloud SDK 139.0.1
alpha 2017.01.12
beta 2017.01.12
bq 2.0.24
bq-nix 2.0.24
core 2017.01.12
core-nix 2017.01.12
gcloud 
gsutil 4.22
gsutil-nix 4.22

```
Followed cloud SDK install instructions at https://cloud.google.com/sdk/docs/quickstart-debian-ubuntu

**Comment by chanseokoh** [comment]
Hi @jeffdahl,

One possibility is that the `gcloud` executed on the command-line is different from `gcloud` set and used in Eclipse. What does it return when you do `which gcloud` on the command-line? What is the directory you set in _Preferences_ --> _Google Cloud Tools_ --> _Google Cloud SDK Location_?

![selection_002](https://cloud.githubusercontent.com/assets/10523105/21960082/2052f434-daab-11e6-880f-d8ea6ddf49a7.png)

If they are same, then this needs more investigation.

**Comment by jeffdahl** [comment]
Thanks for the quick response.  

Unfortunately, 
```
jeff@page:~$ which gcloud
/usr/bin/gcloud
```
_Preferences --> Google Cloud Tools --> Google Cloud SDK Location_
![image](https://cloud.githubusercontent.com/assets/6977031/21960351/ed0ad0fa-da9c-11e6-892d-d9be984a3e2a.png)

`/usr/bin/gcloud` resolves to:
```
jeff@page:~$ ls -l /usr/bin/gcloud
lrwxrwxrwx 1 root root 34 Jan 12 16:20 /usr/bin/gcloud -> ../lib/google-cloud-sdk/bin/gcloud
```

the same place.  It is the same cloud sdk.

**Comment by elharo** [comment]
I am unable to immediately reproduce this on my Mac. Which version of Cloud Tools for Eclipse do you have installed?

**Comment by chanseokoh** [comment]
@jeffdahl, this is because a file `/usr/lib/google-cloud-sdk/VERSION` doesn't exist, which we read to check the version. I filed an [issue](https://github.com/GoogleCloudPlatform/appengine-plugins-core/issues/298).

A workaround would be to download one of the [versioned Cloud SDK archives](https://cloud.google.com/sdk/downloads#versioned). Simply unzip an archive, e.g., under your home directory, and set the Cloud SDK path in Eclipse to the unzipped folder. Don't forget to add the `bin` directory of the unzipped home at the front of your PATH, so that you don't accidentally execute `/usr/bin/gcloud`.

**Comment by chanseokoh** [comment]
@jeffdahl I believe this is fixed since gcloud "141.0.0".

**Comment by jeffdahl** [comment]
Awesome, thanks.  I was able to work around it by creating the version file
you mentioned.

Thanks again,
Jeff
